src/index.d.ts

```diff
@@ -178,7 +178,7 @@ export class Processor {
 
 export interface Config {
     // used to match objects when diffing arrays, by default only === operator is used
-    objectHash?: (item: any, index: number) => string;
+    objectHash?: (item: any, index: number, position: 'left' | 'right') => string;
```

Comment:
I wonder if `index` and `position` should be optional, as they are not always set -> https://github.com/contentful/jsondiffpatch/blob/master/src/filters/arrays.js#L72